packages/mobile/src/store/root.saga.ts

```diff
@@ -1,19 +1,29 @@
-import { all, takeEvery } from 'typed-redux-saga'
+import { all, takeEvery, takeLeading, fork, cancelled } from 'typed-redux-saga'
 import { nativeServicesMasterSaga } from './nativeServices/nativeServices.master.saga'
 import { navigationMasterSaga } from './navigation/navigation.master.saga'
 import { initMasterSaga } from './init/init.master.saga'
 import { initActions } from './init/init.slice'
 import { setupCryptoSaga } from './init/setupCrypto/setupCrypto.saga'
 import { publicChannels } from '@quiet/state-manager'
 import { showNotificationSaga } from './nativeServices/showNotification/showNotification.saga'
+import { clearReduxStore } from './nativeServices/leaveCommunity/leaveCommunity.saga'
 
 export function* rootSaga(): Generator {
-  yield all([
-    takeEvery(initActions.setStoreReady.type, setupCryptoSaga),
-    takeEvery(initActions.setStoreReady.type, initMasterSaga),
-    takeEvery(initActions.setStoreReady.type, navigationMasterSaga),
-    takeEvery(initActions.setStoreReady.type, nativeServicesMasterSaga),
-    // Below line is reponsible for displaying notifications about messages from channels other than currently viewing one
-    takeEvery(publicChannels.actions.markUnreadChannel.type, showNotificationSaga),
-  ])
+  console.log('rootSaga starting')
+  try {
+    yield all([
+      fork(setupCryptoSaga),
+      fork(initMasterSaga),
+      fork(navigationMasterSaga),
+      fork(nativeServicesMasterSaga),
```

Comment:
Running these multiple times without cancelling and restarting `rootSaga` results in multiple sagas being run for each event. I didn't see any need to cancel and restart `rootSaga` like we do with `useIO`, so I just used `fork` here.